cli/module_loader.rs

```diff
@@ -1,45 +1,139 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
 
+use crate::cache::EmitCache;
+use crate::emit::emit_parsed_source;
 use crate::emit::TsTypeLib;
+use crate::graph_util::ModuleEntry;
 use crate::proc_state::ProcState;
+use crate::text_encoding::source_map_from_code;
 
+use deno_ast::MediaType;
+use deno_core::anyhow::anyhow;
 use deno_core::error::AnyError;
 use deno_core::futures::future::FutureExt;
 use deno_core::futures::Future;
+use deno_core::resolve_url;
 use deno_core::ModuleLoader;
+use deno_core::ModuleSource;
 use deno_core::ModuleSpecifier;
+use deno_core::ModuleType;
 use deno_core::OpState;
+use deno_core::SourceMapGetter;
 use deno_runtime::permissions::Permissions;
 use std::cell::RefCell;
 use std::pin::Pin;
 use std::rc::Rc;
 use std::str;
 
+pub struct ModuleCodeSource {
+  pub code: String,
+  pub map: Option<String>,
+  pub found_url: ModuleSpecifier,
+  pub media_type: MediaType,
+}
+
 pub struct CliModuleLoader {
   pub lib: TsTypeLib,
   /// The initial set of permissions used to resolve the static imports in the
   /// worker. They are decoupled from the worker (dynamic) permissions since
   /// read access errors must be raised based on the parent thread permissions.
   pub root_permissions: Permissions,
   pub ps: ProcState,
+  // the emit cache can only be used on one thread due to sqlite constraints
+  emit_cache: EmitCache,
```

Comment:
I had to move a lot out of proc_state and into this file because `EmitCache` can only be used on a single thread whereas ProcState gets sent around all over the place.